arches/app/utils/i18n.py

```diff
@@ -58,8 +93,9 @@ def get_localized_value(obj, lang=None, return_lang=False):
 
             if settings.LANGUAGE_CODE in obj:
                 found_lang = settings.LANGUAGE_CODE
-            else:
```

Comment:
The original code basically translated to: "if I can't match any of the keys in the dictionary to a language code, assign the last key in the dict to `found_lang`. If this is called on a non-i18n dictionary it fails silently and in an interesting way. Instead I've removed this logic and replaced it with throwing an error if this `get_localized_value` is called on a non-i18n dict